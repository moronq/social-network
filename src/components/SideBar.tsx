import React from 'react'
import { Link } from 'react-router-dom'

const SideBar = () => {
  return (
    <aside>
      <ul className="flex flex-col w-36 gap-0.5 cursor-pointer">
        <li className="px-2 rounded-md hover:bg-slate-100 ">
          <Link to="/profile">My profile</Link>
        </li>
        <li className="px-2 rounded-md hover:bg-slate-100">
          <Link to="/message">Messages</Link>
        </li>
        <li className="px-2 rounded-md hover:bg-slate-100">
          <Link to="/users">Users</Link>
        </li>
      </ul>
    </aside>
  )
}

export default SideBar
