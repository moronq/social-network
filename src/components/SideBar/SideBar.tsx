import { Link } from 'react-router-dom'

const SideBar = () => {
  return (
    <aside>
      <ul className="flex flex-col w-36 gap-0.5 cursor-pointer">
        <li>
          <Link to="/profile">
            <div className="px-2 rounded-md hover:bg-slate-100 ">
              My profile
            </div>
          </Link>
        </li>
        <li>
          <Link to="/message">
            <div className="px-2 rounded-md hover:bg-slate-100">Messages</div>
          </Link>
        </li>
        <li>
          <Link to="/users">
            <div className="px-2 rounded-md hover:bg-slate-100">Users</div>
          </Link>
        </li>
      </ul>
    </aside>
  )
}

export default SideBar
