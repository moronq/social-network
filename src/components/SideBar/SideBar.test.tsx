import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import SideBar from './SideBar'

describe('SideBar test group', () => {
  test('SideBar renders without error', () => {
    render(
      <MemoryRouter>
        <SideBar />
      </MemoryRouter>
    )
    expect(screen.getByText('My profile')).toBeInTheDocument
    expect(screen.getByText('Messages')).toBeInTheDocument
    expect(screen.getByText('Users')).toBeInTheDocument
    expect(screen.getByRole('list')).toBeInTheDocument
    expect(screen.getAllByRole('link')).toBeInTheDocument
  })
  test('SideBar snapshot', () => {
    const sideBar = render(
      <MemoryRouter>
        <SideBar />
      </MemoryRouter>
    )
    expect(sideBar).toMatchSnapshot()
  })
})
