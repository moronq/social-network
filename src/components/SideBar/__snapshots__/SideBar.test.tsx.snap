// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar test group SideBar snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside>
        <ul
          class="flex flex-col w-36 gap-0.5 cursor-pointer"
        >
          <li>
            <a
              href="/profile"
            >
              <div
                class="px-2 rounded-md hover:bg-slate-100 "
              >
                My profile
              </div>
            </a>
          </li>
          <li>
            <a
              href="/message"
            >
              <div
                class="px-2 rounded-md hover:bg-slate-100"
              >
                Messages
              </div>
            </a>
          </li>
          <li>
            <a
              href="/users"
            >
              <div
                class="px-2 rounded-md hover:bg-slate-100"
              >
                Users
              </div>
            </a>
          </li>
        </ul>
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside>
      <ul
        class="flex flex-col w-36 gap-0.5 cursor-pointer"
      >
        <li>
          <a
            href="/profile"
          >
            <div
              class="px-2 rounded-md hover:bg-slate-100 "
            >
              My profile
            </div>
          </a>
        </li>
        <li>
          <a
            href="/message"
          >
            <div
              class="px-2 rounded-md hover:bg-slate-100"
            >
              Messages
            </div>
          </a>
        </li>
        <li>
          <a
            href="/users"
          >
            <div
              class="px-2 rounded-md hover:bg-slate-100"
            >
              Users
            </div>
          </a>
        </li>
      </ul>
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
